src/main/CommandHandler.cpp

```diff
@@ -768,6 +770,107 @@ CommandHandler::scpInfo(std::string const& params, std::string& retStr)
     retStr = root.toStyledString();
 }
 
+Json::Value
+makeQuorumSetNode(stellar::SCPQuorumSet const& value) {
+    Json::Value node;
+    node["threshold"] = value.threshold;
+
+    Json::Value validators;
+    for (auto const& key : value.validators) {
+        validators.append(KeyUtils::toShortString(key));
+    }
+    node["validators"] = std::move(validators);
+
+    Json::Value innerSets;
+    for (auto const& set : value.innerSets) {
+        innerSets.append(makeQuorumSetNode(set));
+    }
+    node["innerSets"] = std::move(innerSets);
+
+    return node;
+}
+
+void
+CommandHandler::showConfig(std::string const& params, std::string& retStr)
+{
+    Json::Value root;
+
+    auto addVector = [&root](std::string const& name, std::vector<std::string> const& values) {
+        Json::Value vector;
+        for (auto const& elem : values) {
+            vector.append(elem);
+        }
+        root[name] = std::move(vector);
+    };
+
+    Config const& config = mApp.getConfig();
+
+    root["CURRENT_LEDGER_PROTOCOL_VERSION"] = config.CURRENT_LEDGER_PROTOCOL_VERSION;
+    root["FORCE_SCP"] = config.FORCE_SCP;
+    root["RUN_STANDALONE"] = config.RUN_STANDALONE;
+    root["MANUAL_CLOSE"] = config.MANUAL_CLOSE;
+    root["CATCHUP_COMPLETE"] = config.CATCHUP_COMPLETE;
+    root["CATCHUP_RECENT"] = config.CATCHUP_RECENT;
+    root["MAINTENANCE_ON_STARTUP"] = config.MAINTENANCE_ON_STARTUP;
+    root["ARTIFICIALLY_GENERATE_LOAD_FOR_TESTING"] = config.ARTIFICIALLY_GENERATE_LOAD_FOR_TESTING;
+    root["ARTIFICIALLY_ACCELERATE_TIME_FOR_TESTING"] = config.ARTIFICIALLY_ACCELERATE_TIME_FOR_TESTING;
+    root["ARTIFICIALLY_SET_CLOSE_TIME_FOR_TESTING"] = config.ARTIFICIALLY_SET_CLOSE_TIME_FOR_TESTING;
+    root["ARTIFICIALLY_PESSIMIZE_MERGES_FOR_TESTING"] = config.ARTIFICIALLY_PESSIMIZE_MERGES_FOR_TESTING;
+    root["ALLOW_LOCALHOST_FOR_TESTING"] = config.ALLOW_LOCALHOST_FOR_TESTING;
+    root["FAILURE_SAFETY"] = config.FAILURE_SAFETY;
+    root["UNSAFE_QUORUM"] = config.UNSAFE_QUORUM;
+    root["LEDGER_PROTOCOL_VERSION"] = config.LEDGER_PROTOCOL_VERSION;
+    if (config.PREFERRED_UPGRADE_DATETIME) {
+        root["PREFERRED_UPGRADE_DATETIME"] = VirtualClock::tmToISOString(*config.PREFERRED_UPGRADE_DATETIME);
+    }
+    root["OVERLAY_PROTOCOL_MIN_VERSION"] = config.OVERLAY_PROTOCOL_MIN_VERSION;
+    root["OVERLAY_PROTOCOL_VERSION"] = config.OVERLAY_PROTOCOL_VERSION;
+    root["VERSION_STR"] = config.VERSION_STR;
+    root["LOG_FILE_PATH"] = config.LOG_FILE_PATH;
+    root["BUCKET_DIR_PATH"] = config.BUCKET_DIR_PATH;
+    root["DESIRED_BASE_FEE"] = config.DESIRED_BASE_FEE;
+    root["DESIRED_BASE_RESERVE"] = config.DESIRED_BASE_RESERVE;
+    root["DESIRED_MAX_TX_PER_LEDGER"] = config.DESIRED_MAX_TX_PER_LEDGER;
+    root["HTTP_PORT"] = config.HTTP_PORT;
+    root["PUBLIC_HTTP_PORT"] = config.PUBLIC_HTTP_PORT;
+    root["HTTP_MAX_CLIENT"] = config.HTTP_MAX_CLIENT;
+    std::string NETWORK_PASSPHRASE; // identifier for the network
```

Comment:
we can safely output that value too - it is not a secret, it is more a network identifier

Reply:
Done.